src/lua/debug.lua

```diff
@@ -0,0 +1,8 @@
+local debug = require('debug')
+
+local function sourcepath()
+    return debug.getinfo(2, "S").source:sub(2):match("(.*/)") or './'
```

Comment:
We need to elaborate and document what will going on in the following cases:

* foo.lua requires sub/bar.lua and calls a function from it that calls debug.sourcepath().
* foo.lua requires sub/bar.lua and calls a function from it that creates a fiber that calls debug.sourcepath().

Quick test shows that the result will be './' in the former case, while './sub' in the latter one.

Reply:
You are probably ran into a pitfall, when a function like this:

```lua
function()
    return debug.sourcepath()
end
```
will not appear in a stacktrace. In that case your intention may mismatch with what function actually does (`debug.getinfo(2, ...)`). We have to add this note into documentation

Reply:
I double checked. This was because of tail call optimization. This use case already covered in commit message. This warning will also go to the docs.